lib/jnpr/junos/factcache.py

```diff
@@ -211,6 +212,22 @@ def __repr__(self):
                     string = current
         return '{' + string + '}'
 
+    def __repr__(self):
+        """
+        A formated string representation of the facts dictionary.
+
+        :returns string: a formated string representation of the dictionary.
+          Because this returns the value of every fact, it has the
+          side-effect of causing any ungathered facts to be gathered and then
+          cached.
+        """
+        # below 3 line will help in populating _cache. Otherwise pprint
+        # sometime return empty dictionary
+        for key in sorted(self):
+            if not key.startswith('_'):
+                self.get(key)
+        return pformat(self._cache)
```

Comment:
Your implementation causes the "internal" facts (the facts whose names begin with an underscore) to also be printed. In addition, I believe the whole method implementation can be reduced to:
```
return pformat(dict(self))
```